2021_CommandBased/src/main/java/frc/robot/util/XdrivePoseEstimator.java

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) FIRST and other WPILib contributors.
```

Comment:
Can you summarize the difference between this and the default PoseEstimators?

Reply:
the default PoseEstimators are all drivetrain specific, so this one works with the other xdrive classes

Reply:
Could the kinematics class inherit from an existing kinematics class? Maybe that would remove the need for extra copies.

Reply:
That would probably be the right solution, but I didn't feel like redoing all of the classes that would need this. I also expect this class to need further changes, and it's easier to understand the code when it's all in one place.